Tasks/VsTestPlatformToolInstaller/vstestplatformtoolInstaller.ts

```diff
@@ -0,0 +1,160 @@
+                tl.debug('Looking for latest available version in cache.');
+                testPlatformVersion = 'x';
+            }
+            else {
+                tl.debug(`Found the latest version to be ${testPlatformVersion}`);
+            }
+        } catch(error) {
+            // Failed to list available versions, look for the latest version available in the cache
+            tl.warning(tl.loc('FailedToListAvailablePackagesFromNuget'));
+            tl.debug('Looking for latest available version in cache.');
+            testPlatformVersion = 'x';
+        }
+    }
+
+    // Check cache for the specified version
+    tl.debug(`Looking for version ${testPlatformVersion} in the tools cache.`);
+    toolPath = toolLib.findLocalTool('VsTest', testPlatformVersion);
+
+    if ((!toolPath || toolPath === 'undefined')) {
+        if(testPlatformVersion && testPlatformVersion !== 'x') {
+            tl.debug(`Could not find ${packageName}.${testPlatformVersion} in the tools cache. Fetching it from nuget.`);
+            if (toolLib.isExplicitVersion(testPlatformVersion)) {
+                // Download the required version and cache it
+                try {
+                    toolPath = await acquireAndCacheVsTestPlatformNuget(testPlatformVersion);
+                } catch(error) {
+                    // Download failed, look for the latest version available in the cache
+                    tl.warning(tl.loc('TestPlatformDownloadFailed', testPlatformVersion));
+                    testPlatformVersion = 'x';
+                    toolPath = toolLib.findLocalTool('VsTest', testPlatformVersion);
+                    if(!toolPath || toolPath === 'undefined') {
+                        // No version found in cache, fail the task
+                        tl.warning(tl.loc('NoPackageFoundInCache'));
+                        throw new Error(tl.loc('FailedToAcquireTestPlatform'));
+                    }
+                }
+            }
+            else {
+                throw new Error(tl.loc('ProvideExplicitVersion', testPlatformVersion));
+            }
+        } else {
+            tl.warning(tl.loc('NoPackageFoundInCache'));
+            throw new Error(tl.loc('FailedToAcquireTestPlatform'));
+        }
+    }
+
+    // Set the task variable so that the VsTest task can consume this path
+    tl.setVariable('VsTestToolsInstallerInstalledToolLocation', toolPath);
+    console.log(tl.loc('InstallationSuccessful', toolPath));
+    tl.debug('Set variable VsTestToolsInstallerInstalledToolLocation value to ' + toolPath);
+}
+
+function getLatestPackageVersionNumber(includePreRelease: boolean): string {
+    const nugetTool = tl.tool(path.join(__dirname, 'nuget.exe'));
+
+    if(includePreRelease === true) {
+        nugetTool.line('list ' + packageName + ' -PreRelease' + ' -Source ' + packageSource);
+    }
+    else {
+        nugetTool.line('list ' + packageName + ' -Source ' + packageSource);
```

Comment:
telemetry for failures, where we find ultimately, version we find and perf times